underfs/swift/src/main/java/tachyon/underfs/swift/direct/SwiftDirectUnderFileSystem.java

```diff
@@ -0,0 +1,400 @@
+  /** JOSS access object */
+  private final Access mAccess;
+
+  public SwiftDirectUnderFileSystem(String containerName,
+      TachyonConf tachyonConf) throws Exception {
+    super(tachyonConf);
+    LOG.debug("Constructor init: {}", containerName);
+    AccountConfig config = new AccountConfig();
+    config.setUsername(tachyonConf.get(Constants.SWIFT_USER_KEY));
+    config.setTenantName(tachyonConf.get(Constants.SWIFT_TENANT_KEY));
+    config.setPassword(tachyonConf.get(Constants.SWIFT_API_KEY));
+    config.setAuthUrl(tachyonConf.get(Constants.SWIFT_AUTH_URL_KEY));
+    String authMethod = tachyonConf.get(Constants.SWIFT_AUTH_METHOD_KEY);
+    if (authMethod != null && authMethod.equals("keystone")) {
+      config.setAuthenticationMethod(AuthenticationMethod.KEYSTONE);
+    } else {
+      config.setAuthenticationMethod(AuthenticationMethod.TEMPAUTH);
+    }
+
+    ObjectMapper mapper = new ObjectMapper();
+    mapper.configure(SerializationConfig.Feature.WRAP_ROOT_VALUE, true);
+    mContainerName = containerName;
+    mAccount = new AccountFactory(config).createAccount();
+    mAccess = mAccount.authenticate();
+    LOG.debug("Check if container: {} exists ", containerName);
+    Container containerObj = mAccount.getContainer(containerName);
+    if (!containerObj.exists()) {
+      containerObj.create();
+    }
+    mContainerPrefix = Constants.HEADER_SWIFT + mContainerName
+        + PATH_SEPARATOR;
+  }
+
+  @Override
+  public void close() throws IOException {
+    LOG.debug("close");
+  }
+
+  @Override
+  public void connectFromMaster(TachyonConf conf, String hostname) {
+    LOG.debug("connect from master");
+  }
+
+  @Override
+  public void connectFromWorker(TachyonConf conf, String hostname) {
+    LOG.debug("connect from worker");
+  }
+
+  @Override
+  public OutputStream create(String path) throws IOException {
+    LOG.debug("Create method: " + path);
+    String newPath = path.substring(Constants.HEADER_SWIFT.length());
+    SwiftDirectOutputStream out = SwiftDirectClient.PUT(mAccess, newPath);
+    return out;
+  }
+
+  @Override
+  public OutputStream create(String path,
+      int blockSizeByte) throws IOException {
+    LOG.warn("Create with block size is not supported"
+        + "with SwiftDirectUnderFileSystem. Block size will be " + "ignored.");
+    return create(path);
+  }
+
+  @Override
+  public OutputStream create(String path, short replication, int blockSizeByte) throws IOException {
+    LOG.warn("Create with block size and replication is not"
+        + "supported with SwiftDirectUnderFileSystem."
+        + " Block size and replication will be ignored.");
+    return create(path);
+  }
+
+  /* (non-Javadoc)
+   * @see tachyon.underfs.UnderFileSystem#delete(java.lang.String, boolean)
+   *
+   * recursive parameter is not being used for delete.
+   * The reason is that each path is treated as a single object, which
+   * was created with 'create' method.
+   *
+   * Method return 'true' always.
```

Comment:
So the delete semantics is different from S3 or basic `tachyon.underfs.UnderFileSystem#delete`.  Note that, this may break some assumption because Tachyon may have or guarantee to the upper applications. (@calvinjia , right?) Could you leave a TODO here and we could open JIRA to address this separately.